hydrolib/tools/ext_old_to_new/converters.py

```diff
@@ -131,29 +190,77 @@ def convert(self, forcing: ExtOldForcing) -> Boundary:
-            in an old external forcings file. This object contains all the
-            necessary information, such as quantity, values, and timestamps,
-            required for the conversion process.
+                in an old external forcings file. This object contains all the
+                necessary information, such as quantity, values, and timestamps,
+                required for the conversion process.
+            start_time:
+                The start date of the time series data.
 
         Returns:
             Boundary: A Boundary object that represents the converted forcing
             block, ready to be included in a new external forcings file. The
             Boundary object conforms to the new format specifications, ensuring
-            compatibility with updated systems anbd models.
+            compatibility with updated systems and models.
 
         Raises:
             ValueError: If the forcing block contains a quantity that is not
             supported by the converter, a ValueError is raised. This ensures
             that only compatible forcing blocks are processed, maintaining
             data integrity and preventing errors in the conversion process.
+
+        Notes:
+            - The `root_dir` property must be set before calling this method.
+            - Since the `start_time` argument must be provided from the mdu file to convert the time series data,
+            boundary Condition can be only converted by reading the mdu file and the external forcing file is not
+            enough.
         """
+        from hydrolib.core.dflowfm.polyfile.models import PolyFile
+
+        location_file = forcing.filename.filepath
+        poly_line = forcing.filename
+        if not isinstance(poly_line, PolyFile):
+            poly_line = PolyFile(location_file)
+
+        num_files = poly_line.number_of_points
+        if self.root_dir is None:
+            raise ValueError(
+                "The 'root_dir' property must be set before calling this method."
+            )
+
+        tim_files = [
+            self.root_dir
+            / poly_line.filepath.with_name(
+                f"{poly_line.filepath.stem}_000{i + 1}.tim"
+            ).name
+            for i in range(num_files)
+        ]
+
+        tim_model = self.parse_tim_model(tim_files, forcing)
+        # switch the quantity names from the Tim model (loction names) to quantity names.
+        user_defined_names = tim_model.quantities_names
+        tim_model.quantities_names = [forcing.quantity] * len(tim_model.get_units())
+
+        units = tim_model.get_units()
+
+        forcing_model = self.convert_tim_to_bc(
+            tim_model, start_time, units=units, user_defined_names=user_defined_names
+        )
+        # set the bc file names to the same names as the tim files.
+        for i, file in enumerate(forcing_model):
+            file.filepath = tim_files[i].with_suffix(".bc").name
```

Comment:
Will be nicer if all .tim files are converted and collected in a single .bc file (at least for all [Boundary] objects together).